stl/inc/xutility

```diff
@@ -4084,36 +4115,44 @@ struct _False_copy_cat {
     static constexpr bool _Trivially_copyable = false;
 };
 
-template <class _Source, class _Dest>
-struct _Ptr_copy_cat : _False_copy_cat {}; // unwrap the pointer-like type and dispatch to _Ptr_cat_helper for copy
+// NOTE: pointer is not a contiguous iterator if it points to volatile type
+template <class _Source, class _Dest, bool _Both_contiguous = _Iterators_are_contiguous<_Source, _Dest>>
+struct _Ptr_move_cat : _False_copy_cat {};

```

Comment:
Conventionally we use the primary template for the `true` case and a partial specialization for the `false` case when we have metaprogramming that selects a class template specialization on the value of a `bool` non-type template parameter, absent a technical reason to do otherwise. (Also on 4130.)

Reply:
Done